Composer/packages/lib/bot-deploy/src/botProjectDeploy.ts

```diff
@@ -0,0 +1,597 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { WebSiteManagementClient } from '@azure/arm-appservice-profile-2019-03-01-hybrid';
+import { ResourceManagementClient } from '@azure/arm-resources';
+import {
+  Deployment,
+  DeploymentsCreateOrUpdateResponse,
+  DeploymentsValidateResponse,
+  ResourceGroup,
+  ResourceGroupsCreateOrUpdateResponse,
+} from '@azure/arm-resources/esm/models';
+import { GraphRbacManagementClient } from '@azure/graph';
+import * as msRestNodeAuth from '@azure/ms-rest-nodeauth';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import * as rp from 'request-promise';
+import * as util from 'util';
+import { BotProjectDeployConfig } from './botProjectDeployConfig';
+import archiver = require('archiver');
+const exec = util.promisify(require('child_process').exec);
+const luBuild = require('@microsoft/bf-lu/lib/parser/lubuild/builder.js');
+const { promisify } = require('util');
+const readdir = promisify(fs.readdir);
+
+export class BotProjectDeploy {
+  private subId: string;
+  private creds: any;
+  private projPath: string;
+  private deploymentSettingsPath: string;
+  private deployFilePath: string;
+  private zipPath: string;
+  private publishFolder: string;
+  private settingsPath: string;
+  private templatePath: string;
+  private dotnetProjectPath: string;
+  private generatedFolder: string;
+  private remoteBotPath: string;
+  private logger: (string) => any;
+
+  private readonly tenantId = '72f988bf-86f1-41af-91ab-2d7cd011db47';
```

Comment:
whose tenant id is this? It's used globally across all instances of this library?

Reply:
it belongs to microsoft organization, and we need this tenant id to initialize a ms graph client to create an app registration based on the password provided by user.
```
    const graphClient = new GraphRbacManagementClient(credsForGraph, this.tenantId, {
      baseUri: 'https://graph.windows.net',
    });
```

for app registration, azure js sdk also provide an options for whether this application is available to other tenants, 

```
  /**
   * Whether the application is available to other tenants.
   */
  availableToOtherTenants?: boolean;
```

https://github.com/Azure/azure-sdk-for-node/blob/master/Documentation/Authentication.md#for-azure-graph-sdk